src/shogun/lib/parameter_observers/ParameterObserverCV.cpp

```diff
@@ -177,7 +165,12 @@ CrossValidationStorage* CParameterObserverCV::get_observation(int run) const
 	return obs;
 }
 
-const int CParameterObserverCV::get_num_observations() const
+const int32_t CParameterObserverCV::get_num_observations() const
 {
-	return m_observations.size();
+	try {
+		return size_t_to_int32_cast(m_observations.size());
```

Comment:
Exactly in those lines! Would be good to print some debug info on the actual values maybe?